From 7ba6fc7408b5fb46c698e17471d12dd53d5685ec Mon Sep 17 00:00:00 2001
From: reimar <reimar@b3059339-0415-0410-9bf9-f77b7e298cf2>
Date: Sun, 1 Nov 2020 15:07:44 +0000
Subject: [PATCH] ad_spdif: Use avformat_free_context to free context.

This avoids memleaks and in newest FFmpeg versions also
compile errors.

git-svn-id: svn://git.mplayerhq.hu/mplayer/trunk@38199 b3059339-0415-0410-9bf9-f77b7e298cf2
---
 libmpcodecs/ad_spdif.c | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/libmpcodecs/ad_spdif.c b/libmpcodecs/ad_spdif.c
index 1fc237b..8ee4724 100644
--- a/libmpcodecs/ad_spdif.c
+++ b/libmpcodecs/ad_spdif.c
@@ -298,14 +298,8 @@ static void uninit(sh_audio_t *sh)
         if (spdif_ctx->header_written)
             av_write_trailer(lavf_ctx);
         av_freep(&lavf_ctx->pb);
-        if (lavf_ctx->streams) {
-            av_freep(&lavf_ctx->streams[0]->codec);
-            av_freep(&lavf_ctx->streams[0]->info);
-            av_freep(&lavf_ctx->streams[0]);
-        }
-        av_freep(&lavf_ctx->streams);
-        av_freep(&lavf_ctx->priv_data);
+        avformat_free_context(lavf_ctx);
+        lavf_ctx = NULL;
     }
-    av_freep(&lavf_ctx);
     av_freep(&spdif_ctx);
 }
-- 
2.7.4

